### DISCUSSION TITLE
PID control terms

### DISCUSSION THREAD

**Discussion by gySebastian** [tinymovr/Tinymovr#278] [Q&A - Tinymovr] [answered]
Hello, 

I want to ask if it would be possible to control the Derivative term of the pid controller in either position mode or velocity mode to effectively influence the damping effect(resistance of motion). I am not an expert in control theory so excuse me for any lack of understanding. 

Alternatively, is there another way to dynamically change paramters of the controller to easily scale the holding torque of the actuator or the effort in Nm? I am trying to figure out a way to achieve dynamic impedance of the actuator to hold position untill force is excided. Or get it to behave as a "teachable" colaborative robot, where the setpoint can be dragged manualy by pushing the actuator with some X force (Nm). In addition, how would one go about adding a feed-forward force while in position controll?

I hope my questions make sense, and thank you in advance for any suggestions.

**Comment by yconst** [comment, marked as answer]
In position mode you could increase the velocity P term which has almost the same effect as the position D term, only it acts on filtered values of velocity.

> Alternatively, is there another way to dynamically change paramters of the controller to easily scale the holding torque of the actuator or the effort in Nm?

Yes, you can change the control gains and feed-forward values on the fly. Please take a look at the [API reference](https://tinymovr.readthedocs.io/en/latest/protocol/reference.html) and [Control Principles](https://tinymovr.readthedocs.io/en/latest/control/theory.html) for more information.

> Or get it to behave as a "teachable" colaborative robot, where the setpoint can be dragged manualy by pushing the actuator with some X force (Nm)

This is planned to be added in an upcoming firmware update
